com.unity.mobile.notifications/Runtime/Android/AndroidNotificationCenter.cs

```diff
@@ -308,75 +335,133 @@ public static AndroidNotificationIntentData GetLastNotificationIntent()
+            if (!string.IsNullOrEmpty(notification.LargeIcon))
+                s_NotificationManagerClass.CallStatic("setNotificationIcon", notificationBuilder, "largeIcon", notification.LargeIcon);
+            notificationBuilder.Call<AndroidJavaObject>("setContentTitle", notification.Title).Dispose();
+            notificationBuilder.Call<AndroidJavaObject>("setContentText", notification.Text).Dispose();
+            notificationBuilder.Call<AndroidJavaObject>("setAutoCancel", notification.ShouldAutoCancel).Dispose();
+            if (notification.Number >= 0)
+                notificationBuilder.Call<AndroidJavaObject>("setNumber", notification.Number).Dispose();
+            if (notification.Style == NotificationStyle.BigTextStyle)
+            {
+                using (var style = new AndroidJavaObject("android.app.Notification$BigTextStyle"))
+                {
+                    style.Call<AndroidJavaObject>("bigText", notification.Text).Dispose();
+                    notificationBuilder.Call<AndroidJavaObject>("setStyle", style).Dispose();
+                }
+            }
             long timestampValue = notification.ShowCustomTimestamp ? notification.CustomTimestamp.ToLong() : fireTime;
-            notificationIntent.Call<AndroidJavaObject>("putExtra", "timestamp", timestampValue);
+            notificationBuilder.Call<AndroidJavaObject>("setWhen", timestampValue).Dispose();
+            if (!string.IsNullOrEmpty(notification.Group))
+                notificationBuilder.Call<AndroidJavaObject>("setGroup", notification.Group).Dispose();
+            if (notification.GroupSummary)
+                notificationBuilder.Call<AndroidJavaObject>("setGroupSummary", notification.GroupSummary).Dispose();
+            if (!string.IsNullOrEmpty(notification.SortKey))
+                notificationBuilder.Call<AndroidJavaObject>("setSortKey", notification.SortKey).Dispose();
+            notificationBuilder.Call<AndroidJavaObject>("setShowWhen", notification.ShowTimestamp).Dispose();
+            int color = notification.Color.ToInt();
+            if (color != 0)
+                s_NotificationManagerClass.CallStatic("setNotificationColor", notificationBuilder, color);
+            s_NotificationManagerClass.CallStatic("setNotificationUsesChronometer", notificationBuilder, notification.UsesStopwatch);
+            s_NotificationManagerClass.CallStatic("setNotificationGroupAlertBehavior", notificationBuilder, (int)notification.GroupAlertBehaviour);
+
+            using (var extras = notificationBuilder.Call<AndroidJavaObject>("getExtras"))
+            {
+                extras.Call("putInt", "id", id);
+                extras.Call("putLong", "repeatInterval", notification.RepeatInterval.ToLong());
+                extras.Call("putLong", "fireTime", fireTime);
+                if (!string.IsNullOrEmpty(notification.IntentData))
+                    extras.Call("putString", "data", notification.IntentData);
```

Comment:
Can we call it intentData to keep the consistency?

Reply:
This used to be called like that before, so I decided it's better to keep the name.